### PR TITLE
[Port] Void Cloaks, Rune Knives and Claudrons

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Ports Void Cloaks, Rune Knives and Claudrons for heretics to use.

From PR tgstation/tgstation#54252

## Why It's Good For The Game

These items are super robust and may make or break heretics.

## Changelog
:cl:
add: Void Cloaks and Rune Knives are now available for ash heretics.
add: Crucible is now available for rust heretics.
/:cl:

### CODE REVIEW

**Comment by Naevii** [comment]
Good god, this cloak sprite is awfull.

**Comment by utilbird** [comment]
> Good god, this cloak sprite is awfull.

If only we had a talented sprite artist who could improve it

File: code/datums/status_effects/buffs.dm [hunk 1]
```diff
@@ -515,3 +515,81 @@
...
+	REMOVE_TRAIT(owner,TRAIT_IGNOREDAMAGESLOWDOWN,type)
+
+/datum/status_effect/marshal/tick()
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbie = owner
+
+	carbie.adjustFireLoss(carbie.getFireLoss()/5)
+	carbie.adjustBruteLoss(carbie.getBruteLoss()/5)
```
Comment (park66665) [inline code/datums/status_effects/buffs.dm]:
```suggestion
	carbie.adjustFireLoss(carbie.getFireLoss() * -0.2)
	carbie.adjustBruteLoss(carbie.getBruteLoss() * -0.2)
```

Healing through `adjustXLoss` procs requires negative number, and `getXLoss` returns positive number.

File: code/datums/status_effects/buffs.dm [hunk 2]
```diff
@@ -515,3 +515,81 @@
...
+	tick_interval = 1 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/marshal
+
+/datum/status_effect/marshal/on_apply()
+	. = ..()
+	ADD_TRAIT(owner,TRAIT_IGNOREDAMAGESLOWDOWN,type)
+
+/datum/status_effect/marshal/on_remove()
+	. = ..()
+	REMOVE_TRAIT(owner,TRAIT_IGNOREDAMAGESLOWDOWN,type)
```
Comment (park66665) [inline code/datums/status_effects/buffs.dm]:
```suggestion
	REMOVE_TRAIT(owner,TRAIT_IGNOREDAMAGESLOWDOWN,type)
	return ..()
```

Consistency.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 3]
```diff
@@ -237,20 +237,21 @@
...
+				if(21.01 to 41) //.01s are used in case drunkenness ends up to be a small decimal
+					msg += "[t_He] [t_is] flushed.\n"
+				if(41.01 to 51)
+					msg += "[t_He] [t_is] quite flushed and [t_his] breath smells of alcohol.\n"
+				if(51.01 to 61)
+					msg += "[t_He] [t_is] very flushed and [t_his] movements jerky, with breath reeking of alcohol.\n"
+				if(61.01 to 91)
+					msg += "[t_He] look[p_s()] like a drunken mess.\n"
+				if(91.01 to INFINITY)
+					msg += "[t_He] [t_is] a shitfaced, slobbering wreck.\n"
```
Comment (park66665) [inline code/modules/mob/living/carbon/human/examine.dm]: Unrelated changes.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 4]
```diff
@@ -303,12 +304,13 @@
...
+			var/cyberimp_detect
 			for(var/obj/item/organ/cyberimp/CI in internal_organs)
 				if(CI.status == ORGAN_ROBOTIC && !CI.syndicate_implant)
-					cyberimp_detect += CI.name
-			if(length(cyberimp_detect))
-				. += "Detected cybernetic modifications: [english_list(cyberimp_detect)]"
+					cyberimp_detect += "[name] is modified with a [CI.name]."
+			if(cyberimp_detect)
+				. += "Detected cybernetic modifications:"
+				. += cyberimp_detect
```
Comment (park66665) [inline code/modules/mob/living/carbon/human/examine.dm]: ditto.

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 5]
```diff
@@ -179,3 +179,186 @@
...
+	. += "This item can carve 'Grasping carving' - when stepped on it causes heavy damage to the legs and stuns for 5 seconds."
+	. += "This item can carve 'Mad carving' - when stepped on it causes dizzyness, jiterryness, temporary blindness, confusion , stuttering and slurring."
+
+/obj/item/melee/rune_knife/Initialize()
+	. = ..()
+	linked_action = new(src)
+
+/obj/item/melee/rune_knife/Destroy()
+	. = ..()
+	QDEL_NULL(linked_action)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm]:
```suggestion
	QDEL_NULL(linked_action)
	return ..()
```

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 6]
```diff
@@ -179,3 +179,186 @@
...
+
+/obj/item/melee/rune_knife/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!is_type_in_typecache(target,blacklisted_turfs) && !drawing && proximity_flag)
+		carve_rune(target,user,proximity_flag,click_parameters)
+
+///Action of carving runes, gives you the ability to click on floor and choose a rune of your need.
+/obj/item/melee/rune_knife/proc/carve_rune(atom/target, mob/user, proximity_flag, click_parameters)
+	var/obj/structure/trap/eldritch/elder = locate() in range(1,target)
+	if(elder)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm]:
```suggestion
	if(locate(/obj/structure/trap/eldritch) in range(1, target))
```

**Comment by TeomanTheGreat** [comment]
So is this getting merged?

**Review by park66665 — COMMENTED** [review]
I took a look at the code briefly. The code was properly ported... mostly.

**Comment by eeSPee** [comment]
Alright. Merge Conflict should be no more.

**Comment by heepox** [comment]
some day...

**Comment by eeSPee** [comment]
I'm not gonna bother fixing conflicts until it is approved.

**Comment by ItsTisme** [comment]
Quick tip for conflict. get the dmi icons from the upstream commit those, then export the icons for the new sprites and import them into the dmi files and then another commit

**Comment by ike709** [comment]
I'm of the opinion that heretics don't need any further buffs, but I'm not super up-to-date on the state of heretics so if another maintainer disagrees they can reopen.

**Comment by zeskorion** [comment]
> I'm of the opinion that heretics don't need any further buffs, but I'm not super up-to-date on the state of heretics so if another maintainer disagrees they can reopen.

They get validhunted to death as soon as someone sees a pierced reality afaik. Its more of a player issue
